Source/player.h

```diff
@@ -555,6 +556,20 @@ struct Player {
 	 */
 	int GetManaShieldDamageReduction();
 
+	/**
+	 * @brief Gets the effective spell level for the player, considering item bonuses
+	 * @param spell spell_id enum member identifying the spell
+	 * @return effective spell level
+	 */
+	int GetSpellLevel(spell_id spell) const
+	{
+		if (spell == SPL_INVALID || static_cast<std::size_t>(spell) >= sizeof(_pSplLvl)) {

```

Comment:
I'm not sure we really need this check when already limiting it to `spell_id`, we could possibly define `_pSplLvl` as `_pSplLvl[SPL_LAST]` instead.

```suggestion
		if (spell == SPL_INVALID) {
```

Reply:
With the way enums are handled in C++ any 6 bit value is valid for spell_id and because the backing type is specified potentially any 8 bit value could be passed in to this function without triggering undefined behaviour. This is overly defensive for the current use case given it's called with a known value, it's more for future handling of loaded hotkeys given there's no validation performed on the way in.